src/main/java/ua/danit/final_project/controllers/security/AuthenticationController.java

```diff
@@ -0,0 +1,105 @@
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import ua.danit.final_project.configuration.security.JwtAuthenticationRequest;
+import ua.danit.final_project.configuration.security.JwtAuthenticationResponse;
+import ua.danit.final_project.configuration.security.JwtTokenUtil;
+import ua.danit.final_project.controllers.exceptions.AuthenticationException;
+import ua.danit.final_project.dto.DefaultMapper;
+import ua.danit.final_project.dto.UserDto;
+import ua.danit.final_project.entities.User;
+import ua.danit.final_project.services.RegistrationService;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Objects;
+
+@RestController
+public class AuthenticationController {
+
+  @Value("${jwt.header}")
+  private String tokenHeader;
+
+  private final AuthenticationManager authenticationManager;
+  private final JwtTokenUtil jwtTokenUtil;
+  private final UserDetailsService userDetailsService;
+  private final RegistrationService registrationService;
+  private final DefaultMapper mapper;
+
+  public AuthenticationController(AuthenticationManager authenticationManager,
+                                  JwtTokenUtil jwtTokenUtil,
+                                  @Qualifier("jwtUserDetailsService") UserDetailsService userDetailsService,
+                                  RegistrationService registrationService,
+                                  DefaultMapper mapper) {
+    this.authenticationManager = authenticationManager;
+    this.jwtTokenUtil = jwtTokenUtil;
+    this.userDetailsService = userDetailsService;
+    this.registrationService = registrationService;
+    this.mapper = mapper;
+  }
+
+  @PostMapping("/auth")
+  public ResponseEntity<JwtAuthenticationResponse> createAuthenticationToken(@RequestBody JwtAuthenticationRequest authenticationRequest)
+      throws AuthenticationException {
+
+    authenticate(authenticationRequest.getUserName(), authenticationRequest.getUserPassword());
+
+    final UserDetails userDetails = userDetailsService.loadUserByUsername(authenticationRequest.getUserName());
+    final String token = jwtTokenUtil.generateToken(userDetails);
+
+    return ResponseEntity.ok(new JwtAuthenticationResponse(token));
+  }
+
+  @PostMapping("/register")
+  public ResponseEntity register(@RequestBody User user)
+      throws AuthenticationException {
+
+    registrationService.register(user);
+    return ResponseEntity.ok("New user created");
+  }
+
+  @GetMapping("/refresh")
+  public ResponseEntity<JwtAuthenticationResponse> refreshAndGetAuthenticationToken(HttpServletRequest request) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove usage of generic wildcard type. ([squid:S1452](https://sonarcloud.io/organizations/final-project/rules#rule_key=squid%3AS1452))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=final-project&pullRequest=173&issues=AWb1IgCVW3Y69bz5KWF0&open=AWb1IgCVW3Y69bz5KWF0)

[ISSUE_KEY]: AWb1IgCVW3Y69bz5KWF0